### PR TITLE
feat/add-prev-next-btns-to-lessons

### LINKED ISSUE #17509: Feature Request : Add 'previous' and 'next' buttons for navigating challenges.
#### Describe your problem and - if possible - how to reproduce it
I completed some of the challenges a few months ago. Now, I wanted to revisit those challenges to have a look at some of the concepts. My previously submitted solutions are not available anymore. So I can't move to next challenge without submitting a new solution again which I don't want to do. I can only go to next challenge from 'Curriculum' which is really inefficient.
The 'Next' button could be placed between the search bar and  'Curriculum'.

#### Tell us about your browser and operating system

* Browser Name: Google Chrome
* Browser Version: 67
* Operating System: Ubuntu 18.04

### PR DESCRIPTION
**Pull request by moT01**
I'm not sure if this is even wanted, as there wasn't much discussion on the issue about it. I think they're pretty slick, so I hope this feature/PR is taken into consideration.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/17509

**================================================================** 

**In order for this to work in testing, I believe you need to delete the hidden `client/.cache` folder before running `npm run develop`.**

**================================================================**

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
@moT01 so forward work as expected.
back takes me to undefined.
load the page with a new browser, go to map while signed in. go to the first challenge. then go forward and go backward. you will get undefined.

**Comment by raisedadead** [comment]
Hi @moT01 

Thanks for the PR. 

We are currently under a feature freeze. I have left some explanation here: https://gitter.im/FreeCodeCamp/Contributors?at=5c7a23172ca5ec547462f85d

Will add more comprehensive communication in the an related issue thread soon.

**Review by thecodingaviator — APPROVED** [review]
Just tested it locally, awesome work! 🎊

**Review by ojeytonwilliams — APPROVED** [review]
I also gave it a whirl, LGTM!  I couldn't recreate the bug when going to previous challenges - it works just fine for me.

**Comment by RandellDawson** [comment]
The bug I mentioned in the issue is no longer present.

**Comment by raisedadead** [comment]
The community needs a feature --> we love the community --> let's do it right. 

Opening up for QA. Thanks @moT01 for the PR and everyone for playing with this.

**Review by RandellDawson — APPROVED** [review]

**Comment by thecodingaviator** [comment]
We have three approvals. @QuincyLarson do you wanna take a look?

**Comment by raisedadead** [comment]
> We have three approvals...

Just to be clear the dev-team QA is pending.

**Comment by ojeytonwilliams** [comment]
Just to confirm what @moT01 said, the previous button linking to undefined only happens if you try to use the cache of a build without the buttons.

To reproduce:
* Rebase to master
* checkout master
* delete `client/.cache`
* `npm run develop`
* checkout the rebased branch
* `npm run develop`

In short: definitely not a bug.

File: client/src/templates/Challenges/components/challenge-title.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+.challenge-title {
+  display: inline;
+  padding: 0 15px;
+  margin: 0;
+  flex-grow: 2;
+}
+
+.challenge-title-wrap > Button {
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/challenge-title.css]:
```suggestion
```
It looks a bit weird when the buttons resize, so I'd recommend leaving them fixed.

File: client/src/templates/Challenges/components/challenge-title.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  align-items: center;
+  justify-content: space-around;
+  margin-bottom: 1.45rem;
+}
+
+.challenge-title {
+  display: inline;
+  padding: 0 15px;
+  margin: 0;
+  flex-grow: 2;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/challenge-title.css]:
```suggestion
```
Seems redundant with the button `flex-grow` change below.

File: client/src/templates/Challenges/components/challenge-title.css [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+.challenge-title-wrap > Button {
+  flex-grow: 1;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/challenge-title.css]:
```suggestion
```

File: client/src/templates/Challenges/components/challenge-title.css [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+}
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/components/challenge-title.css]:
```suggestion
```

**Review by ojeytonwilliams — COMMENTED** [review]
While playing around with it, I found the buttons resizing to be a bit odd:
ResizingButtons

Perhaps it's better if they don't change.

**Comment by moT01** [comment]
Thanks for opening this up @raisedadead - and all the reviews everyone! I added @ojeytonwilliams suggestion to make the buttons a static size - as well as add a few properties to stop the buttons from jumping around so much like @RandellDawson mentioned in the issue

File: client/src/templates/Challenges/components/Challenge-Title.js [hunk 5]
```diff
@@ -15,10 +30,25 @@ function ChallengeTitle({ children, isCompleted }) {
     );
   }
   return (
-    <h2 className='text-center challenge-title'>
-      {children || 'Happy Coding!'}
-      {icon}
-    </h2>
+    <div className='challenge-title-wrap'>
+      {showPrevNextBtns ? (
+        <Button bsStyle='primary' onClick={() => navigate(prevChallengePath)}>
```
Comment (raisedadead) [inline client/src/templates/Challenges/components/Challenge-Title.js]:
```suggestion
        <Button
          aria-label='Previous lesson'
          bsStyle='primary'
          className='btn-invert'
          onClick={() => navigate(prevChallengePath)}
        >
```

File: client/src/templates/Challenges/components/Challenge-Title.js [hunk 6]
```diff
@@ -15,10 +30,25 @@ function ChallengeTitle({ children, isCompleted }) {
...
+          &lt;
+        </Button>
+      ) : null}
+      <h2 className='text-center challenge-title'>
+        {children || 'Happy Coding!'}
+        {icon}
+      </h2>
+      {showPrevNextBtns ? (
+        <Button
```
Comment (raisedadead) [inline client/src/templates/Challenges/components/Challenge-Title.js]:
```suggestion
        <Button
          aria-label='Next lesson'
          bsStyle='primary'
          className='btn-invert'
          onClick={() => navigate(introPath ? introPath : nextChallengePath)}
        >
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
Lets, make the button inverted. Please note that we want primary to be a call to action. And that should usually be one action item. It's already associated with run tests.

Next is the ability to be a11y complaint. Those `>` && `<` would hurt a11y. I have added some labels in the suggestion to make it complaint.

I am inclined that we should use text links instead of buttons, but that can come later.

**Review by raisedadead — COMMENTED** [review]
Also some tests would be awesome, this is a great feature. Let's make it resilient.

**Comment by moT01** [comment]
I think I need some help coming up with tests for this - I took a look and couldn't come up with anything. Anyone want to work with me on it? or take a look or something?

**Comment by ojeytonwilliams** [comment]
I'd be happy to work on this with you.  I'll be around on Gitter tomorrow.

**Comment by ahmadabdolsaheb** [comment]
@raisedadead

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
Nice work @moT01 on working on this. And thanks to everyone who tested this.

**Comment by moT01** [comment]
yay! 🎉 - thanks for merging this and all the reviews everyone!

**Pull request merged**